holmes/plugins/toolsets/kubernetes_logs.py

```diff
@@ -0,0 +1,284 @@
+            # Try to load the kubeconfig and access the API
+            self._initialize_client()
+            self._api_client.call_api(
+                "/version",
+                "GET",
+                auth_settings=["BearerToken"],
+                response_type="object",
+                _return_http_data_only=True,
+            )
+            return True, ""
+        except Exception as e:
+            return False, f"Kubernetes client initialization error: {str(e)}"
+
+    def get_example_config(self):
+        return LoggingConfig().model_dump()
+
+    def _initialize_client(self):
+        if not self._api_client:
+            try:
+                try:
+                    config.load_incluster_config()
+                except config.ConfigException:
+                    config.load_kube_config()
+
+                self._api_client = client.ApiClient()
+                self._core_v1_api = client.CoreV1Api(self._api_client)
+            except Exception:
+                logging.error("Failed to initialize Kubernetes client", exc_info=True)
+                raise
+
+    def fetch_logs(self, params: FetchLogsParams) -> StructuredToolResult:
+        try:
+            self._initialize_client()
+
+            pod = self._find_pod(params.namespace, params.pod_name)
+            all_logs = []
+            if not pod or not pod.containers:
+                all_logs = self._fetch_pod_logs(
+                    params=params, previous=True
+                ) + self._fetch_pod_logs(params=params, previous=False)
+            else:
+                all_logs = self._fetch_pod_logs(
+                    params=params, containers=pod.containers, previous=True
+                )
+                all_logs = all_logs + self._fetch_pod_logs(
+                    params=params, containers=pod.containers, previous=False
+                )
+
+            if params.filter_pattern:
+                all_logs = self._filter_logs(all_logs, params.filter_pattern)
+
+            if params.limit and params.limit < len(all_logs):
+                all_logs = all_logs[-params.limit :]
+
+            # Format output
+            formatted_logs = self._format_logs(all_logs)
+
+            return StructuredToolResult(
+                status=ToolResultStatus.SUCCESS,
+                data=formatted_logs,
+                params=params.model_dump(),
+            )
+        except ApiException as e:
+            return StructuredToolResult(
+                status=ToolResultStatus.ERROR,
+                error=f"Kubernetes API error: {str(e)}",
+            )
+        except Exception as e:
+            logging.exception(f"Error fetching logs for pod {params.pod_name}")
+            return StructuredToolResult(
+                status=ToolResultStatus.ERROR,
+                error=f"Error fetching logs: {str(e)}",
+            )
+
+    def _find_pod(self, namespace: str, pod_name: str) -> Optional[Pod]:
+        if not self._core_v1_api:
```

Comment:
Not sure i fully understand. In which case we can get to here and the _core_v1_api isn't initialized? 
If it is safe guard i think that we atleast need to add here warning log.

Reply:
Yes this should really not be necessary but it's due to how the toolsets have been designed to take the config. It's a tech debt we've been dragging for a long time. I'll add the log